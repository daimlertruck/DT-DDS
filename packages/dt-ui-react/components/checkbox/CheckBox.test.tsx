import { withProviders } from '@dt-ui/react-core';
import { render } from '@testing-library/react';
import React from 'react';

import { Checkbox } from '.';

describe('<CheckBox /> component', () => {
  const ProvidedCheckBox = withProviders(Checkbox);

  it('renders checkbox with label', () => {
    const { container } = render(
      <ProvidedCheckBox
        isChecked={false}
        isDisabled={false}
        onChange={() => null}
      >
        Label
      </ProvidedCheckBox>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders checkbox without label', () => {
    const { container } = render(
      <ProvidedCheckBox
        isChecked={false}
        isDisabled={false}
        onChange={() => null}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders checkbox checked', () => {
    const { container } = render(
      <ProvidedCheckBox isChecked isDisabled={false} onChange={() => null} />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders checkbox disabled', () => {
    const { container } = render(
      <ProvidedCheckBox isChecked={false} isDisabled onChange={() => null} />
    );
    expect(container).toMatchSnapshot();
  });
});
