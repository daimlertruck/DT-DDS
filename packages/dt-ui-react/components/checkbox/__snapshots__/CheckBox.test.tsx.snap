// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> component renders checkbox checked 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  border: 1px solid #F8F9FA;
  background-color: #FFFFFF;
  -webkit-transform: translateY(-0.075em);
  -moz-transform: translateY(-0.075em);
  -ms-transform: translateY(-0.075em);
  transform: translateY(-0.075em);
  height: 16px;
  width: 16px;
  display: grid;
  place-content: center;
  cursor: pointer;
}

.emotion-2:before {
  content: '';
  width: 0.85em;
  height: 0.85em;
  -webkit-clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  box-shadow: inset 1em 1em #FFFFFF;
  background-color: CanvasText;
}

.emotion-2:checked::before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2:checked {
  background: #036DC1;
  border: none;
}

.emotion-2:disabled {
  pointer-events: none;
}

.emotion-2:disabled:checked {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked) {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked)~label {
  color: #EBECED;
  pointer-events: none;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  font-weight: 400;
  font-size: 18px;
  line-height: 24.12px;
  letter-spacing: normal;
  margin: 0;
  color: #4F4F4F;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="flex-direction: row;"
  >
    <input
      checked=""
      class="emotion-2 emotion-3"
      data-testid="checkbox-id"
      id="checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="checkbox-id"
    />
  </div>
</div>
`;

exports[`<CheckBox /> component renders checkbox disabled 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  border: 1px solid #F8F9FA;
  background-color: #FFFFFF;
  -webkit-transform: translateY(-0.075em);
  -moz-transform: translateY(-0.075em);
  -ms-transform: translateY(-0.075em);
  transform: translateY(-0.075em);
  height: 16px;
  width: 16px;
  display: grid;
  place-content: center;
  cursor: pointer;
}

.emotion-2:before {
  content: '';
  width: 0.85em;
  height: 0.85em;
  -webkit-clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  box-shadow: inset 1em 1em #FFFFFF;
  background-color: CanvasText;
}

.emotion-2:checked::before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2:checked {
  background: #036DC1;
  border: none;
}

.emotion-2:disabled {
  pointer-events: none;
}

.emotion-2:disabled:checked {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked) {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked)~label {
  color: #EBECED;
  pointer-events: none;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  font-weight: 400;
  font-size: 18px;
  line-height: 24.12px;
  letter-spacing: normal;
  margin: 0;
  color: #4F4F4F;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="flex-direction: row;"
  >
    <input
      class="emotion-2 emotion-3"
      data-testid="checkbox-id"
      disabled=""
      id="checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="checkbox-id"
    />
  </div>
</div>
`;

exports[`<CheckBox /> component renders checkbox with label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  border: 1px solid #F8F9FA;
  background-color: #FFFFFF;
  -webkit-transform: translateY(-0.075em);
  -moz-transform: translateY(-0.075em);
  -ms-transform: translateY(-0.075em);
  transform: translateY(-0.075em);
  height: 16px;
  width: 16px;
  display: grid;
  place-content: center;
  cursor: pointer;
}

.emotion-2:before {
  content: '';
  width: 0.85em;
  height: 0.85em;
  -webkit-clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  box-shadow: inset 1em 1em #FFFFFF;
  background-color: CanvasText;
}

.emotion-2:checked::before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2:checked {
  background: #036DC1;
  border: none;
}

.emotion-2:disabled {
  pointer-events: none;
}

.emotion-2:disabled:checked {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked) {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked)~label {
  color: #EBECED;
  pointer-events: none;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  font-weight: 400;
  font-size: 18px;
  line-height: 24.12px;
  letter-spacing: normal;
  margin: 0;
  color: #4F4F4F;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="flex-direction: row;"
  >
    <input
      class="emotion-2 emotion-3"
      data-testid="checkbox-id"
      id="checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="checkbox-id"
    >
      Label
    </label>
  </div>
</div>
`;

exports[`<CheckBox /> component renders checkbox without label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  border: 1px solid #F8F9FA;
  background-color: #FFFFFF;
  -webkit-transform: translateY(-0.075em);
  -moz-transform: translateY(-0.075em);
  -ms-transform: translateY(-0.075em);
  transform: translateY(-0.075em);
  height: 16px;
  width: 16px;
  display: grid;
  place-content: center;
  cursor: pointer;
}

.emotion-2:before {
  content: '';
  width: 0.85em;
  height: 0.85em;
  -webkit-clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  clip-path: polygon(28% 38%, 41% 53%, 75% 24%, 86% 38%, 40% 78%, 15% 50%);
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  box-shadow: inset 1em 1em #FFFFFF;
  background-color: CanvasText;
}

.emotion-2:checked::before {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2:checked {
  background: #036DC1;
  border: none;
}

.emotion-2:disabled {
  pointer-events: none;
}

.emotion-2:disabled:checked {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked) {
  background-color: #F8F9FA;
}

.emotion-2:disabled:not(checked)~label {
  color: #EBECED;
  pointer-events: none;
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  font-weight: 400;
  font-size: 18px;
  line-height: 24.12px;
  letter-spacing: normal;
  margin: 0;
  color: #4F4F4F;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="flex-direction: row;"
  >
    <input
      class="emotion-2 emotion-3"
      data-testid="checkbox-id"
      id="checkbox-id"
      type="checkbox"
    />
    <label
      class="emotion-4 emotion-5"
      for="checkbox-id"
    />
  </div>
</div>
`;
